src/import_edf.jl

```diff
@@ -204,22 +208,49 @@ can be either a `String` giving the generated channel name, or a `Pair` mapping
 a canonical name to a list of alternatives that it should be substituted for
 (e.g., `"canonical_name" => ["alt1", "alt2", ...]`).
 
+`unit_alternatives` lists standardized unit names and alternatives that map to them.
+See `OndaEDF.STANDAR_UNITS` for defaults.
+
+`preprocess_labels(label::String)` is applied to raw edf signal header labels
+beforehand; defaults to `identity`.
+
 See `OndaEDF.STANDARD_LABELS` for the labels (`signal_names => channel_names`
 `Pair`s) that are used to extract EDF signals by default.
 
 """
-function extract_channels_by_label(edf::EDF.File, signal_names, channel_names)
+function extract_channels_by_label(edf::EDF.File, signal_names, channel_names; unit_alternatives=STANDARD_UNITS, preprocess_labels=identity)
     matcher = x -> begin
         # yo I heard you like closures
-        return s -> match_edf_label(s.header.label, signal_names,
-                                    x isa Pair ? first(x) : x,
-                                    channel_names)
+        return s -> begin
+            m = match_edf_label(preprocess_labels(s.header.label),
+                                signal_names,
+                                x isa Pair ? first(x) : x,
+                                channel_names)
+            !isnothing(m) && return m
+            # channel info is sometimes misplaced in transducer_type field; only accept these if corresponding signal_name occurs in channel label
+            m = match_edf_label(preprocess_labels(s.header.transducer_type),
+                                signal_names,
+                                x isa Pair ? first(x) : x,
+                                channel_names)
+            isnothing(m) && return m
+            lowercase_label = _safe_lowercase(s.header.label)
+            any(occursin(name, lowercase_label) for name in signal_names) && return m
```

Comment:
can you explain the situations where this fallback is appropriate?  and given that the logic of this closure has become more complex, maybe it should be rolled into `match_edf_label` or another helper function?

Reply:
Using the `transducer_type` as a label is a little dangerous, because all kinds of garbage can show up in the `transducer_type` field which could spuriously match some signal channels; this only accepts a successful match using the transducer type if the label mentions the corresponding signal type.

This whole `transducer_type -> label` fallback is the messiest part of this PR, I don't like it. The main reason I put it there is because it would be annoying to provide a custom extractor to do this, unless I'm overlooking some appropriate design. Lemme count how many test EDF signals would be affected without this.